testing/simulated/payload_cache_test.go

```diff
@@ -0,0 +1,1066 @@
+		s.Reth.LogBuffer.Reset()
+		processResp, processErr = s.Reth.SimComet.Comet.ProcessProposal(s.Reth.CtxComet, processRequest)
+		s.Require().NoError(processErr)
+		s.Require().Equal(types.PROCESS_PROPOSAL_STATUS_ACCEPT, processResp.Status)
+
+		// Finalize the block. Evict bkit payload here because finalize is accepted.
+		finalizeRequest := &types.FinalizeBlockRequest{
+			Txs:             proposal.Txs,
+			Height:          nextBlockHeight,
+			ProposerAddress: rethNodeAddress,
+			Time:            consensusTime,
+		}
+		_, finalizeErr := s.Geth.SimComet.Comet.FinalizeBlock(s.Geth.CtxComet, finalizeRequest)
+		s.Require().NoError(finalizeErr)
+		_, finalizeErr = s.Reth.SimComet.Comet.FinalizeBlock(s.Reth.CtxComet, finalizeRequest)
+		s.Require().NoError(finalizeErr)
+
+		// Commit the block.
+		_, err := s.Geth.SimComet.Comet.Commit(s.Geth.CtxComet, &types.CommitRequest{})
+		s.Require().NoError(err)
+		s.Geth.LogBuffer.Reset()
+		_, err = s.Reth.SimComet.Comet.Commit(s.Reth.CtxComet, &types.CommitRequest{})
+		s.Require().NoError(err)
+		s.Reth.LogBuffer.Reset()
+	}
+
+	// Finally, we cross the fork and show no issues. Geth uses the optimistic build which has the
+	// correct payload time and consequently is built correctly for post-fork.
+	nextBlockHeight++
+	time.Sleep(100 * time.Millisecond) // The optimistic build for next height should have completed by now.
+	{
+		consensusTime := time.Unix(int64(s.Geth.TestNode.ChainSpec.ElectraForkTime()), 0)
+		proposal, prepareErr := s.Geth.SimComet.Comet.PrepareProposal(s.Geth.CtxComet, &types.PrepareProposalRequest{
+			Height:          nextBlockHeight,
+			Time:            consensusTime,
+			ProposerAddress: gethNodeAddress,
+		})
+		s.Require().NoError(prepareErr)
+		s.Require().Len(proposal.Txs, 2)
+
+		processRequest := &types.ProcessProposalRequest{
+			Txs:             proposal.Txs,
+			Height:          nextBlockHeight,
+			ProposerAddress: gethNodeAddress,
+			Time:            consensusTime,
+		}
+		// Process the proposal.
+		processResp, processErr := s.Geth.SimComet.Comet.ProcessProposal(s.Geth.CtxComet, processRequest)
+		s.Require().NoError(processErr)
+		s.Require().Equal(types.PROCESS_PROPOSAL_STATUS_ACCEPT.String(), processResp.Status.String())
+		s.Require().Contains(s.Geth.LogBuffer.String(), "Processing execution requests")
+		processResp, processErr = s.Reth.SimComet.Comet.ProcessProposal(s.Reth.CtxComet, processRequest)
+		s.Require().NoError(processErr)
+		s.Require().Equal(types.PROCESS_PROPOSAL_STATUS_ACCEPT, processResp.Status)
+		s.Require().Contains(s.Reth.LogBuffer.String(), "Processing execution requests")
+	}
+}
+
+// Test a scenario where reth must rebuild a payload for a failed state transition.
+//
+// NOTE: this test requires reth with the --engine.always-process-payload-attributes-on-canonical-head flag.
+func (s *PayloadCacheSuite) TestReth_MustRebuildForFailedStateTransition_IsSuccessful() {
```

Comment:
Note that this test is the same as `TestReth_MultiplePayloadRebuilds` from [here](https://github.com/berachain/beacon-kit/pull/2956/files#diff-c9033c103fe3561778a91428bd7d498f3cbd9a19a3bb88caf339feb13a45d011R210), just renamed and refactored for a 2-node chain